lib/nx_zlib.h

```diff
@@ -336,6 +338,45 @@ typedef struct nx_stream_s {
 } nx_stream;
 typedef struct nx_stream_s *nx_streamp;
 
+/* metric used to decide if it's time to switch to sw, as we are not able
+   to switch mid-stream we cannot use avg_delay directly.  */
+extern uint64_t	metric;
+/* average delay for a nx job */
+extern uint64_t	avg_delay;
+extern uint64_t nx_ticks_per_sec;
+/* decaying average of past delays
+   larger decay value retains the average longer */
+static const uint64_t decay = 4;
+
+/* For estimating device throughput */
+static inline int nx_device_stats(uint64_t start, uint64_t end)
+{
+	/* delay of the last job */
+	uint64_t last_dly = nx_time_diff(start, end);
+
+	/* ignore if the measured delay is too long; either the timer
+	   is broken or the process slept */
+	if (last_dly > nx_ticks_per_sec ||
+	    (last_dly < nx_ticks_per_sec/10000000UL)) {
+		prt_err("Unrealiable delay: %lu tps: %lu\n", last_dly, nx_ticks_per_sec);
+		return -1; /* unreliable */
```

Comment:
This function is called in 2 places, but none of them check for the return.
Does it really need to return an integer?

Reply:
Right, I will remove this.